src/renderer/atlas/AtlasEngine.r.cpp

```diff
@@ -66,7 +66,7 @@ try
     const til::rect fullRect{ 0, 0, _r.cellCount.x, _r.cellCount.y };
 
     // A change in the selection or background color (etc.) forces a full redraw.
-    if (WI_IsFlagSet(_r.invalidations, RenderInvalidations::ConstBuffer))
+    if (WI_IsFlagSet(_r.invalidations, RenderInvalidations::ConstBuffer) || _r.customPixelShader)

```

Comment:
_huh_. what happened without this?

Reply:
It would only present the given dirty area, which for a custom shader might not include all pixels that changed.